odoo/http.py

```diff
@@ -456,6 +456,16 @@ def is_cors_preflight(request, endpoint):
     return request.httprequest.method == 'OPTIONS' and endpoint.routing.get('cors', False)
 
 
+def get_context(exception):
+    current = exception
+    while current:
+        ctx = getattr(current, "context", None)
+        if ctx:  # return as soon as context is found
+            return ctx
+        current = current.__cause__
```

Comment:
With this you won't have to change the `raise ... from e` (and if you do, there are multiple places where you should change it in that file).
```suggestion
        current = current.__cause__ or current.__context__
```

Reply:
As we are creating new access error instance so I think it much needed `from e`. And in that file all access errors are already coming `from e`.